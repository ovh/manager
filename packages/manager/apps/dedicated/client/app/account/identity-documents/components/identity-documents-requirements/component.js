import template from './template.html';

export default {
  template,
};
