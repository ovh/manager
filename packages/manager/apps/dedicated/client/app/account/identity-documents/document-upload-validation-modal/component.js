import template from './template.html';

export default {
  template,
  bindings: {
    onClose: '&',
    isOpenModal: '<',
  },
};
