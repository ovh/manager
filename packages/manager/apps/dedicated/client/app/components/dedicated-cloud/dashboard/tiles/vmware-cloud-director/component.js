import controller from './controller';
import template from './template.html';

export default {
  bindings: {
    goToVcdOrder: '<',
    productId: '<',
    trackingPrefix: '<',
    pccMigrationState: '<',
  },
  name: 'ovhManagerPccDashboardVmwareCloudDirector',
  controller,
  template,
};
