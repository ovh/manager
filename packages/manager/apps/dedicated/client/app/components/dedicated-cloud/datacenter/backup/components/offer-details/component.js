import template from './template.html';

export default {
  bindings: {
    offer: '<',
  },
  template,
};
