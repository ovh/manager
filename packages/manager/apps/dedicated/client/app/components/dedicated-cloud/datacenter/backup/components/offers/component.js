import controller from './controller';
import template from './template.html';

export default {
  bindings: {
    backup: '<',
    backupOffers: '<',
    currentOfferLabel: '<',
    previewMode: '<',
    user: '<',
    disabled: '<',
  },
  controller,
  template,
};
