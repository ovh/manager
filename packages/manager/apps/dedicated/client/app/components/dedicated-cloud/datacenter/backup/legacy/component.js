import controller from './controller';
import template from './template.html';

export default {
  bindings: {
    backupTariffUrl: '<',
    enabledBackupOffer: '<',
    trackingPrefix: '<',
  },
  controller,
  template,
};
