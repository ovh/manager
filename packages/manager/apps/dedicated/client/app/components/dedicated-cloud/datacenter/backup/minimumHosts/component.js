import template from './template.html';

export default {
  bindings: {
    goToOrderHosts: '<',
    minimumHosts: '<',
  },
  template,
};
