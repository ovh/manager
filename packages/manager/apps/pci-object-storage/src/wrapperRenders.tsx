import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import {
  ShellContext,
  ShellContextType,
} from '@ovh-ux/manager-react-shell-client';
import { vi } from 'vitest';
import { createMemoryRouter, RouterProvider } from 'react-router-dom';
import React from 'react';

export const shellContext = {
  environment: {
    getUser: () => ({ ovhSubsidiary: 'mocked_ovhSubsidiary' }),
  },
  shell: {
    navigation: {
      getURL: vi.fn(),
    },
    tracking: {
      trackPage: vi.fn(),
      trackClick: vi.fn(),
    },
  },
};

const queryClient = new QueryClient({
  defaultOptions: {
    mutations: {
      retry: false,
    },
    queries: {
      retry: false,
    },
  },
});

export const wrapper = ({ children }) => (
  <QueryClientProvider client={queryClient}>
    <ShellContext.Provider
      value={(shellContext as unknown) as ShellContextType}
    >
      {children}
    </ShellContext.Provider>
  </QueryClientProvider>
);

export const mockShellContext = {
  environment: {
    getRegion: vi.fn().mockReturnValue('EU'),
    getUser: vi.fn().mockReturnValue({ ovhSubsidiary: 'mocked_ovhSubsidiary' }),
    getApplication: vi.fn(),
  },
  shell: {
    environment: {
      getRegion: vi.fn().mockReturnValue('EU'),
      getUser: vi.fn().mockReturnValue({}),
      getApplication: vi.fn(),
    },
    navigation: {
      getURL: vi.fn(),
    },
    tracking: {
      trackPage: vi.fn(),
      trackClick: vi.fn(),
    },
  },
};

export const wrapperShow = ({ children }: { children: React.ReactNode }) => {
  const router = createMemoryRouter(
    [
      {
        path: '/',
        element: children,
        handle: {
          crumb: () => 'Home',
        },
      },
    ],
    {
      initialEntries: ['/'],
      future: {
        v7_startTransition: true,
        v7_relativeSplatPath: true,
      } as any,
    },
  );

  return (
    <QueryClientProvider client={queryClient}>
      <ShellContext.Provider value={mockShellContext as any}>
        <RouterProvider router={router} />
      </ShellContext.Provider>
    </QueryClientProvider>
  );
};

export const wrapperOffsiteReplication = ({
  children,
}: {
  children: React.ReactNode;
}) => (
  <QueryClientProvider client={queryClient}>
    <ShellContext.Provider value={mockShellContext as any}>
      {children}
    </ShellContext.Provider>
  </QueryClientProvider>
);
