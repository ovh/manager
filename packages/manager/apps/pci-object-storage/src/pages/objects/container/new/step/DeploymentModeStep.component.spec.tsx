import { describe } from 'vitest';
import { render } from '@testing-library/react';
import { DeploymentModeStep } from './DeploymentModeStep.component';
import { wrapper } from '@/wrapperRenders';

describe('DeploymentModeStep', () => {
  it('should display correctly', () => {
    const { container } = render(<DeploymentModeStep />, {
      wrapper,
    });
    expect(container).toMatchSnapshot();
  });
});
