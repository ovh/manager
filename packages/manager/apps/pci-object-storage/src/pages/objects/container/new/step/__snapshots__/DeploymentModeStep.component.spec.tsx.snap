// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DeploymentModeStep > should display correctly 1`] = `
<div>
  <section
    class="flex flex-row border-0 border-t-[1px] border-solid border-t-[--ods-color-neutral-100] pt-5 mb-5"
  >
    <div
      class="basis-[40px]"
    >
      <div
        class="flex justify-center items-center font-bold border-2 border-solid rounded-full h-10 w-10 border-[--ods-color-neutral-500]"
      >
        <span
          class="font-semibold text-[16px] leading-[20px] text-[--ods-color-neutral-500]"
        >
          2
        </span>
      </div>
    </div>
    <div
      class="basis-full px-5"
    >
      <div
        class="flex flex-col md:flex-row"
      >
        <div
          class="font-normal leading-[43px] p-0 m-0 w-full md:w-5/6 text-[20px] text-[--ods-color-neutral-500]"
        >
          pci_projects_project_storages_containers_deployment_mode_title
        </div>
      </div>
    </div>
  </section>
</div>
`;
